File: 01 clase entero/js/entero.js
class Entero {
    // Atributos
    Num;

    // Constructor = inicializa el objeto
    constructor(Numero) {
        this.Num = Numero;
    }

    // CLASS - Cargar el valor de Num desde un input HTML
    setNum() {
        const input = document.getElementById("numeroInput");
        this.Num = parseInt(input.value, 10);
    }

    // CLASS - Obtener el valor de Num
    getNum() {
        return this.Num;
    }

    // CLASS - Mostrar el valor de Num
    showNum() {
        const resultadoDiv = document.getElementById('resultado');
        resultadoDiv.textContent = this.getNum();
    }

    showResultado(respuesta) {
        const resultadoDiv = document.getElementById('resultado');
        resultadoDiv.textContent = respuesta;
    }

    IncrementarNum() {
        this.Num = this.Num + 1;
    }

    DecrementarNum() {
        this.Num = this.Num - 1;
    }

    esParImpar() {
        return (this.Num % 2 == 0);
    }

    esPositivoNegativo() {
        if (this.Num > 0) {
            return "Es positivo";
        } else if (this.Num < 0) {
            return "Es negativo";
        } else {
            return "Es cero";
        }
    }

    esPerfecto() {
        var suma = 0;
        for (let i = 1; i < this.Num; i++) {
            if (this.Num % i == 0) {
                suma += i;
            }
        }
        return suma === this.Num;
    }

    factorial() {
        var factorial = 1;
        for (let i = 1; i <= this.Num; i++) {
            factorial *= i;
        }
        return factorial;
    }

    // Método para calcular los primeros n números de Fibonacci
    Fibonacci() {
        let n = this.Num; // Usa el valor de `this.Num` como el número de elementos
        if (n <= 0) {
            return [];
        }

        let fibSecuencia = [];
        let a = 0, b = 1;

        for (let i = 0; i < n; i++) {
            fibSecuencia.push(a);
            let temp = a;
            a = b;
            b = temp + b;
        }

        return fibSecuencia;
    }
}

// Crear el objeto
var ClaseEntero = new Entero(0);

// Las funciones = botones HTML
function cargarNum() {
    ClaseEntero.setNum();
}

function mostrarNum() {
    ClaseEntero.showNum();
}

// Incrementar el valor de Num
function IncrementarValor() {
    ClaseEntero.IncrementarNum();
    ClaseEntero.showNum();
}

// Decrementar el valor de Num
function DecrementarValor() {
    ClaseEntero.DecrementarNum();
    ClaseEntero.showNum();
}

// Verifica si es par o impar
function esParImparNum() {
    var respuesta = ClaseEntero.esParImpar();
    var resp = respuesta ? "El número es Par" : "El número es Impar";
    ClaseEntero.showResultado(resp);
}

// Verificar si es positivo o negativo
function esPositivoNegativoNum() {
    var respuesta = ClaseEntero.esPositivoNegativo();
    ClaseEntero.showResultado(respuesta);
}

// Verificar si es perfecto
function esPerfectoNum() {
    var respuesta = ClaseEntero.esPerfecto();
    var resp = respuesta ? "El número es Perfecto" : "El número no es Perfecto";
    ClaseEntero.showResultado(resp);
}

// Calcular el factorial
function factorialNum() {
    var respuesta = ClaseEntero.factorial();
    ClaseEntero.showResultado(respuesta);
}

// Mostrar la secuencia de Fibonacci con el número de elementos especificado por el usuario
function fibonacciNum() {
    ClaseEntero.setNum(); // Asegúrate de actualizar el número límite
    var respuesta = ClaseEntero.Fibonacci();
    ClaseEntero.showResultado(respuesta.join(", ")); // Convierte el array a una cadena de texto
}

File: 01 clase entero/py/entero.py
class Entero:
    def __init__(self, Numero):
        self.Num = Numero

    def setNum(self):
        input_value = input("Enter a number: ")
        self.Num = int(input_value)

    def getNum(self):
        return self.Num

    def showNum(self):
        print(f"El número actual es: {self.getNum()}")

    def incrementarNum(self):
        self.Num += 1

    def decrementarNum(self):
        self.Num -= 1

    def esParImpar(self):
        return self.Num % 2 == 0

    def esPerfecto(self):
        suma = 0
        for i in range(1, self.Num):
            if self.Num % i == 0:
                suma += i
        return suma == self.Num

    def esPrimo(self):
        if self.Num <= 1:
            return False
        if self.Num <= 3:
            return True
        if self.Num % 2 == 0 or self.Num % 3 == 0:
            return False
        i = 5
        while i * i <= self.Num:
            if self.Num % i == 0 or self.Num % (i + 2) == 0:
                return False
            i += 6
        return True

    def fibonacci(self):
        limit = self.Num
        if limit < 0:
            return []
        
        fib_secuencia = []
        a, b = 0, 1
        
        while a <= limit:
            fib_secuencia.append(a)
            a, b = b, a + b
        
        return fib_secuencia

    def menu(self):
        while True:
            print("\nMenu")
            print("1. Establecer un nuevo número")
            print("2. Mostrar el número actual")
            print("3. Incrementar el número")
            print("4. Decrementar el número")
            print("5. Verificar si es par o impar")
            print("6. Verificar si es perfecto")
            print("7. Verificar si es primo")
            print("8. Mostrar la secuencia de Fibonacci")
            print("9. Botame")

            opcion = input("Ingrese una opción: ")

            if opcion == "1":
                self.setNum()
            elif opcion == "2":
                self.showNum()
            elif opcion == "3":
                self.incrementarNum()
                self.showNum()
            elif opcion == "4":
                self.decrementarNum()
                self.showNum()
            elif opcion == "5":
                if self.esParImpar():
                    print("El número es par.")
                else:
                    print("El número es impar.")
            elif opcion == "6":
                if self.esPerfecto():
                    print("El número es perfecto.")
                else:
                    print("El número no es perfecto.")
            elif opcion == "7":
                if self.esPrimo():
                    print("El número es primo.")
                else:
                    print("El número no es primo.")
            elif opcion == "8":
                fib_sequence = self.fibonacci()
                print(f"Secuencia de Fibonacci: {fib_sequence}")
            elif opcion == "9":
                print("Me voy, chau pescau. ")
                break
            else:
                print("Opción inválida, intente nuevamente.")

numero = Entero(10)
numero.menu()
